### PR TITLE
Add Modal metrics data to integration

### PR DESCRIPTION
**Pull request by pawalt**
### What does this PR do?

This PR updates the documentation for the Modal integration to include information about metrics, and it adds the reported metrics to metadata.csv.

### Motivation

We are building out the Modal metrics integration for Datadog, and we're listing out our metrics so you know which metrics to expect.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Thanks for the review!

### CODE REVIEW

File: modal/CHANGELOG.md
Comment (eho1307) [inline modal/CHANGELOG.md:7]:
```suggestion
* Container metrics
```

File: modal/README.md
Comment (eho1307) [inline modal/README.md:40]:
```suggestion
See [metadata.csv][3] for a list of metrics included with this integration.
```

**Review by eho1307 — CHANGES_REQUESTED** [review]
A few points of feedback:
1. Could you include these metrics in your dashboard or create a new dashboard to include these metrics? Our dashboards help customers get started quickly. You can follow these [docs](https://docs.datadoghq.com/developers/integrations/create-an-integration-dashboard?tab=ui).
2. Please add the [metrics section](https://docs.datadoghq.com/developers/integrations/check_references/#manifest-file) within the manifest.json. View an example [here](https://github.com/DataDog/integrations-extras/blob/master/ably/manifest.json).

File: modal/README.md
Comment (hestonhoffman) [inline modal/README.md:1]: On line 65, please add the following:
```
[5]: https://github.com/DataDog/integrations-extras/blob/master/modal/metadata.csv
```

File: modal/README.md
Comment (pawalt) [reply modal/README.md:40]: done

File: modal/CHANGELOG.md
Comment (pawalt) [reply modal/CHANGELOG.md:7]: done

File: modal/README.md
Comment (pawalt) [reply modal/README.md:1]: Nice thanks

**Comment by pawalt** [comment]
@eho1307 @hestonhoffman thanks for the feedback - I've added the appropriate metadata, and I've added a new dashboard with a new screenshot. Let me know if there's anything else I can fix up.

**Comment by eho1307** [comment]
Looks good from my perspective. Once we have everyone's approval, we'll merge the PR.

**Comment by pawalt** [comment]
thanks @eho1307!

**Comment by pawalt** [comment]
@hestonhoffman let me know if there's anything I can do to improve this PR - I'm happy to make any changes that would be helpful :)

File: modal/README.md
Comment (janine-c) [inline modal/README.md:40]: Not nitpicking your writing here - I think we need to use exact wording to ensure the "magic phrase" works properly:
```suggestion
See [metadata.csv][5] for a list of metrics provided by this integration.
```

**Review by janine-c — CHANGES_REQUESTED** [review]
Hi, I'm stepping in for Heston while he's OOO. Just made a small suggestion request to ensure the "magic phrase" can really do its magic :)

**Comment by pawalt** [comment]
@janine-c thanks! i've made this change

**Review by janine-c — APPROVED** [review]

**Review by london-wharton — APPROVED** [review]

**Comment by pawalt** [comment]
I think we just need ecosystems here - is there someone there who can take a look? Maybe @bgoldberg122 or @james-eichelbaum?

**Comment by eho1307** [comment]
Hi @pawalt we'll be releasing this tomorrow. No more reviews are needed.

**Comment by pawalt** [comment]
@eho1307 thanks so much!

**Pull request merged**
